mcstatus/querier.py

```diff
@@ -60,91 +56,13 @@ def read_query(self) -> QueryResponse:
         self.connection.write(request)
 
         response = self._read_packet()
-        return QueryResponse.from_connection(response)
-
+        return QueryResponse.build(*self.transform_connection_to_objects(response))
 
-class AsyncServerQuerier(ServerQuerier):
-    def __init__(self, connection: UDPAsyncSocketConnection):
-        # We do this to inform python about self.connection type (it's async)
-        super().__init__(connection)  # type: ignore[arg-type]
-        self.connection: UDPAsyncSocketConnection
-
-    async def _read_packet(self) -> Connection:
-        packet = Connection()
-        packet.receive(await self.connection.read(self.connection.remaining()))
-        packet.read(1 + 4)
-        return packet
+    def transform_connection_to_objects(self, response: Connection) -> tuple[RawQueryResponse, list[str]]:
```

Comment:
I'm not a huge fan of the name of this function. Perhaps something simpler like `read_response` would be enough? The current name is pretty odd. It might even make sense to split the method into two, first for reading the raw response, second for reading the player list.

Reply:
I feel that `_parse_response` is better, because `read_response` can be confused with another existing method `read_query`.

Renamed in 7a883e7377f7ae8ff0506523c1a9f2a6df0853b1.